chain/client/src/chunk_validation.rs

```diff
@@ -562,7 +626,7 @@ impl Client {
     ) -> Result<(), Error> {
         let shard_id = chunk.shard_id();
         let chunk_hash = chunk.chunk_hash();
-        let Some(witness) = self.create_state_witness(prev_chunk_header, chunk)? else {
+        let Some(witness) = self.create_state_witness(prev_chunk_header, chunk, None)? else {
```

Comment:
I am not sure `None` is ok here. Maybe we want to retrieve storage proof for transaction validation somehow in case of shadow validation too.

Reply:
`None` here will result in shadow validation failing, so we need to do something about it
Storage proof for new transactions is only generated by chunk producer, so there is no way for us to retrieve it for every block. Instead I suggest we generate it ourselves here by manually running `prepare_transactions` on transactions from `chunk` passing `record_storage = true`. Would that work?